testomania/src/main/java/com/earth/testomania/home_screen/presentation/HomeScreenViewModel.kt

```diff
@@ -16,7 +17,11 @@ import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
-class HomeScreenViewModel @Inject constructor() : ViewModel() {
+class HomeScreenViewModel @Inject constructor(
+     networkConnectivityObserver: NetworkConnectivityObserver,
+) : ViewModel() {
+
+    val networkObserver = networkConnectivityObserver
```

Comment:
You can add `val` to constructor's param and remove this `network Observer` field and we can avoid field duplication